src/102_binary_tree_level_order_traversal.md

```diff
@@ -0,0 +1,119 @@
+# 102. Binary Tree Level Order Traversal
+
+**<font color=red>level: Medium</font>**
+
+## Description
+```c++
+Given a binary tree, return the level order traversal of its nodes' values. (ie, from left to right, level by level).
+
+For example:
+Given binary tree [3,9,20,null,null,15,7],
+
+    3
+   / \
+  9  20
+    /  \
+   15   7
+
+return its level order traversal as:
+
+[
+  [3],
+  [9,20],
+  [15,7]
+]
+```
+
+## C++ Solutions
+
+### Approach 1 : Recursion
+
+* Time complexity O(n)
+* Space complexity O(n)
+
+```c++
+class Solution {
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
+ * };
+ */
+class Solution {
+public:
+    vector<vector<int>> levelOrder(TreeNode* root) {
+        vector<vector<int>> res;
+        traverse(root, 1, res);
+        return res;
+    }
+    
+    void traverse(TreeNode *root, size_t level, vector<vector<int>> &res) {
```

Comment:
Is this function given by leetcode? If it's not, I think we should not use a reference for a output parameter.

Reply:
The function is given by this way.